src/pudl/validate.py

```diff
@@ -1533,6 +1554,7 @@ def plot_vs_agg(orig_df, agg_df, validation_cases):
         "low_bound": 0.95,
         "data_col": "fuel_mmbtu_per_unit",
         "weight_col": "fuel_consumed_units",
+        "xfail": True,
```

Comment:
It might be worthwhile to define a `BoundsCheckParams` class that all these bounds checks conform to, instead of having them be dictionaries that can be anything. It's not *necessary* for this code to work, though. What do people think?

Reply:
This module is a horror show written long ago, before we knew about Pydantic or data classes or typed dicts. It's long overdue for an overhaul. I don't even know if it should continue existing after we rip out `PudlTabl` and change how the tests and validations access the database.

Reply:
I think what makes the most sense is:

1. Use the PUDL resource schemas to define the data type validations. If e.g. "everything in this column should be >10" that seems like it lives in the schema.
2. More complicated validations should be defined alongside the assets - you can just define an `@asset_check` right in the module, in which case we'll have to add some `load_asset_checks_from_module` calls to `etl/__init__.py`. 
3. We should have utility functions + types related to validations in the `pudl.validations` module.

I don't *think* we need to then reference the more complicated validations alongside the schema in RESOURCE_METADATA.